src/permissions_template.c

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+
+#include "kerneltypes.h"
+#include "kmalloc.h"
+#include "permissions_template.h"
+#include "console.h"
+
+#define MAX_NUMBER_OF_PERMISSIONS 169 // temporary, for now they'll be statically allocated
+
+static uint32_t template_counter = 1;
```

Comment:
I didn't want the user to be confused if they get 0 back, since 0 is often associated with failure. It also helps me keep my code straight, since I check for 0 on return often.